php/Blockonomics.php

```diff
@@ -238,4 +292,267 @@ public function testSetup()
+            $address = $currentAddress[0];
+        } else {
+            $responseObj = $blockonomics->new_address(get_option("blockonomics_callback_secret"), $crypto);
+            if($responseObj->response_code != 200) {
+                exit();
+            }
+            $address = $responseObj->address;
+        }
+        $wc_order = new WC_Order($order_id);
+        $order = $orders[$order_id][$address];
+        // Check if order has not expired yet
+        if ( isset($order['timestamp']) && $order['timestamp'] >= time() - get_option("blockonomics_timeperiod") * 60 ) {
+            $timestamp = $order['timestamp'];
+            $satoshi = $order['satoshi'];
+            $status = $order['status'];
+        }else{
+            // Refresh the timestamp and expected satoshi
+            $timestamp = time();
+            if(get_woocommerce_currency() != 'BTC'){
+                $responseObj = $blockonomics->get_price(get_woocommerce_currency(), $crypto);
+                if($responseObj->response_code != 200) {
+                    exit();
+                }
+                $price = $responseObj->price;
+                $price = $price * 100/(100+get_option('blockonomics_margin', 0));
+            }else{
+                $price = 1;
+            }
+            $satoshi = intval(round(1.0e8*$wc_order->get_total()/$price));
+            $status = -1;
+        }
+        $order = array(
+                'value'              => $wc_order->get_total(),
+                'currency'           => get_woocommerce_currency(),
+                'order_id'           => $order_id,
+                'status'             => $status,
+                'satoshi'            => $satoshi,
+                'crypto'             => $crypto,
+                'timestamp'          => $timestamp,
+                'addr'               => $address,
+                'txid'               => ''
+        );
+
+        $orders[$order_id][$address] = $order;
+        update_option('blockonomics_orders', $orders);
+        update_post_meta($order_id, $crypto .'_address', $address);
+        return $order;
+    }
+
+    // Get the order info by id and crypto
+    public function get_order_info($order_id, $crypto){
+        $order = $this->process_order($order_id, $crypto);
+        header("Content-Type: application/json");
+        exit(json_encode($order));
+    }
+
+    // Process the blockonomics callback
+    public function process_callback($secret, $addr, $status, $value, $txid){
+        $callback_secret = get_option("blockonomics_callback_secret");
+        if ($callback_secret  && $callback_secret == $secret) {
+            $orders = get_option('blockonomics_orders');
+            // fetch the order id by bitcoin address
+            foreach($orders as $id => $order){
+                if(isset($order[$addr])){
+                    $order_id = $id;
+                    break;
+                }
+            }
+            if ($order_id){
```

Comment:
Lots of nested ifs cause confusion in code. A better thing to do here is 
```
if (!isset($order_id))
exit("Error: order not found");
```
and carry on forward without adding an extra if nesting

Reply:
Fixed in 7f904d4